### PR TITLE
AUDIO ATOM: add new audio atom

### PR DESCRIPTION
**Pull request by ajwl**
We want to create a new type of atom that can host a playable audio file. 

What this changes: 
1. Changes made in Content-Atom and Content Api Thrift Models have been added by bumping the dependencies 

2. SBT had to be bumped to fix a bug. Have also updated a few other things in the plugins

3. I've made some small unrelated fixes in Atom workshop. eg allowing Atom descriptions to render as html rather than strings. 

4. The audio atom editor is the basic version where all fields are manually added. There are no validity checks on these fields because they are likely to be replaced with functionality that automatically gets these values. 

Audio atoms can now be created and saved in Dynamo, but they cannot be retrieved from CAPI because the CAPI part of the work has not yet been completed. 

# To Do

- [x] Change deps to actual builds from maven, not snapshots
- [x] Do we need a description, get decision from UX
- [x] Can image be optional? 

# Screenshots 
screen shot 2019-01-11 at 14 45 40

screen shot 2019-01-09 at 16 47 02

# Related PRs: 
https://github.com/guardian/content-atom/pull/125
https://github.com/guardian/content-api-models/pull/155
https://github.com/guardian/content-api/pull/2200

# Dependency updates 
Have taken @regiskuckaertz [branch](https://github.com/guardian/content-atom/pull/125) on Content Atom
1.  `content-atom` which I have published locally and added to:
2. `atom-maker` which I have published locally and added to:
3. `Atom Workshop`

Separately this has also been pushed to the content-api-scala-client
1. `content-atom` which I have published locally and added to:
2. `content api thrift models` (incorporating[ this change](https://github.com/guardian/content-api-models/pull/155)) which I have published locally and added to:
3. `content api scala client` which I have published locally and added to: 
4. `Atom Workshop`

### CODE REVIEW

**Comment by ajwl** [comment]
cf: @regiskuckaertz

File: public/js/constants/atomData.js [hunk 1]
```diff
@@ -90,9 +90,15 @@ export const chart = {
...
+  type: "audio",
+  fullName: "Audio",
+  description: "A small audio player"
+};
+
 
-export const allAtomTypes = [cta, recipe, storyQuestions, quiz, media, qa, guide, profile, timeline, explainer, commonsDivision, chart];
-export const workshopEditableAtomTypes = [cta, storyQuestions, recipe, qa, guide, profile, timeline, explainer, commonsDivision, chart];
+export const allAtomTypes = [cta, recipe, qa, guide, profile, timeline, media, chart, audio,  storyQuestions, quiz, explainer, commonsDivision];
+export const workshopEditableAtomTypes = [cta, storyQuestions, recipe, qa, guide, profile, timeline, explainer, commonsDivision, chart, audio];
```
Comment (ajwl) [inline public/js/constants/atomData.js]: I reordered these to put the most used ones first. This means they come at the top on the UI.

File: public/js/components/AtomEdit/CustomEditors/AudioEditor.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    onUpdate: PropTypes.func.isRequired,
+    onFormErrorsUpdate: PropTypes.func
+  };
+
+  render () {
+    return (
+      <div>
+        <ManagedForm formName="audioEditor" data={this.props.atom} updateData={this.props.onUpdate} onFormErrorsUpdate={this.props.onFormErrorsUpdate} >
+          <ManagedField fieldLocation="title" name="Title">
+            <FormFieldTextInput key={1}/>
```
Comment (regiskuckaertz) [inline public/js/components/AtomEdit/CustomEditors/AudioEditor.js]: Do you need those `key` attributes? React ... 🤦‍♂️

File: public/js/components/AtomEdit/CustomEditors/AudioEditor.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+          </ManagedField>
+          <ManagedField fieldLocation="trackUrl" name="Track url">
+            <FormFieldTextInput key={3}/>
```
Comment (regiskuckaertz) [inline public/js/components/AtomEdit/CustomEditors/AudioEditor.js]: I hope this component supports customizing the `type` attribute: `type="url"`

Comment (regiskuckaertz) [reply public/js/components/AtomEdit/CustomEditors/AudioEditor.js]: ... also `required`

File: public/js/components/AtomEdit/CustomEditors/AudioEditor.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+          </ManagedField>
+          <ManagedField fieldLocation="image" name="Select Image">
+            <FormFieldImageSelect gridUrl="jj" />
```
Comment (regiskuckaertz) [inline public/js/components/AtomEdit/CustomEditors/AudioEditor.js]: Is this opening a frame to browse the Grid? If not, I'd suggest using a text input, as above.

File: public/js/components/AtomEdit/CustomEditors/AudioEditor.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+          </ManagedField>
+          <ManagedField fieldLocation="contentId" name="Content ID">
+            <FormFieldTextInput/>
+          </ManagedField>
+          <ManagedField fieldLocation="duration" name="Duration (seconds)">
+            <FormFieldNumericInput/>
+          </ManagedField>
+
+          <ManagedField fieldLocation="crossPlatformLinks.apple" name="Subscription: Apple Podcasts">
+            <FormFieldTextInput/>
```
Comment (regiskuckaertz) [inline public/js/components/AtomEdit/CustomEditors/AudioEditor.js]: These are all URLs, so I'd add the attribute here too if possible.

File: public/js/constants/atomData.js [hunk 6]
```diff
@@ -90,9 +90,15 @@ export const chart = {
   description: "A variety of different charts"
 };
 
+export const audio = {
+  type: "audio",
+  fullName: "Audio",
+  description: "A small audio player"
```
Comment (regiskuckaertz) [inline public/js/constants/atomData.js]: We may want to ask the audio desk or Chris M. what would a good description be.

File: public/js/constants/atomData.js (hunk 1 above)
Comment (regiskuckaertz) [reply public/js/constants/atomData.js]: 👍

File: public/js/components/AtomEdit/CustomEditors/AudioEditor.js (hunk 2 above)
Comment (ajwl) [reply public/js/components/AtomEdit/CustomEditors/AudioEditor.js]: Oh good point. I can at least use an index. Will check that out

File: public/js/components/AtomEdit/CustomEditors/AudioEditor.js (hunk 3 above)
Comment (ajwl) [reply public/js/components/AtomEdit/CustomEditors/AudioEditor.js]: Sure thing

File: public/js/components/AtomEdit/CustomEditors/AudioEditor.js (hunk 4 above)
Comment (ajwl) [reply public/js/components/AtomEdit/CustomEditors/AudioEditor.js]: Yes it is using a frame to browse the Grid

**Comment by ajwl** [comment]
Have added the 2 requireds and an audio icon

**Comment by regiskuckaertz** [comment]
Great work 🤗

**Pull request merged**